papers/libceed-paper/paper.rst

```diff
@@ -0,0 +1,258 @@
+
+Therefore, in recent years, high-order numerical methods have been widely used in Partial Differential Equation (PDE) solvers, but software packages that provide high-performance implementations have often been special-purpose and intrusive. In contrast, libCEED is light-weight, minimally intrusive, and very versatile. In fact, libCEED offers a purely algebraic interface for matrix-free operator representation and supports run-time selection of implementations tuned for a variety of computational device types, including CPUs and GPUs. libCEED's algebraic interface can unobtrusively be integrated in new and legacy software to provide performance portable interfaces. While libCEED's focus is on high-order finite elements, the approach is algebraic and thus applicable to other discretizations in factored form. libCEED's role, as a lightweight portable library that allows a wide variety of applications to share highly optimized discretization kernels, is illustrated in Fig. :ref:`fig-libCEED-backends`, where a non-exhaustive list of specialized implementations (backends) is provided. libCEED provides a low-level Application Programming Interface (API) for user codes so that applications with their own discretization infrastructure (e.g., those in PETSc :cite:`PETScUserManual`, MFEM :cite:`MFEMlibrary` and Nek5000 :cite:`Nekwebsite`) can evaluate and use the core operations provided by libCEED. GPU implementations are available via pure CUDA :cite:`CUDAwebsite` as well as the OCCA :cite:`OCCAwebsite` and MAGMA :cite:`MAGMAwebsite` libraries. CPU implementations are available via pure C and AVX intrinsics as well as the LIBXSMM :cite:`LIBXSMM` library. libCEED provides a unified interface, so that users only need to write a single source code and can select the desired specialized implementation at run time. Moreover, each process or thread can instantiate an arbitrary number of backends.
+
+.. figure:: libCEEDBackends.png
+
+   The role of libCEED as a lightweight, portable library which provides a low-level API for efficient, specialized implementations. libCEED allows different applications to share highly optimized discretization kernels. :label:`fig-libCEED-backends`
+
```

Comment:
I think we can drop " and C interface" part of the first sentence since we don't really discuss the C API past mentioning that the Python interface is a wrapper for the C library. That would help address the reviewers that were concerned that this talks too much about C.

Reply:
Reworded. I meant we'd introduce the API.